todos/src/App.js

```diff
@@ -1,23 +1,45 @@
-import React from "react";
+import React, { useEffect, Fragment } from "react";
 import "App.css";
 import Navbar from "components/Navbar";
 import Todos from "components/Todos";
-import { Provider } from "react-redux";
-import store from "store";
 import TopContent from "components/TopContent";
 import { ToastContainer } from "react-toastify";
+import { useDispatch } from "react-redux";
+import { getCurrentTodos } from "supabaseData";
+import { addInitialData } from "actions/index";
+import { setSplashLoadingState } from "actions";
+import { useSelector } from "react-redux";
+import SplashScreen from "components/SplashScreen";
 
 function App() {
-  return (
-    <React.Fragment>
-      <Provider store={store}>
+  const splashLoadingState = useSelector(
+    (state) => state.loadingReducers.splashLoadingState
+  );
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    const fetchData = async () => {
+      dispatch(setSplashLoadingState(true));
+      const initialData = await getCurrentTodos();
```

Comment:
Shouldn't we do this in redux-thunk?

Reply:
Yes @hhu-wtag Bhai, my bad. I am shifting these into actions right now.